coursier.bzl

```diff
@@ -125,29 +129,17 @@ def generate_imports(repository_ctx, dep_tree, srcs_dep_tree = None):
                 else:
                     # If not, it's a relative path to the one in output_base/external/$maven/v1/...
                     artifact_relative_path = _normalize_to_unix_path(artifact_path)
-                target_label = _escape(_strip_packaging_and_classifier(artifact["coord"]))
+                target_label = _escape(_strip_packaging_classifier(artifact["coord"]))
                 srcjar_paths[target_label] = artifact_relative_path
 
     # Iterate through the list of artifacts, and generate the target declaration strings.
     for artifact in dep_tree["dependencies"]:
         artifact_path = artifact["file"]
+        target_label = _escape(_strip_packaging_classifier(artifact["coord"]))
```

Comment:
instead of using the artifact path as the key to dedupe target generation, we use the target label as the key, since it's canonical and unique anyway